gitlab/pkg/commands/issues.go

```
@@ -0,0 +1,278 @@
+	if err != nil {
+		return gitlab.UpdateIssueOptions{}, err
+	}
+	err = json.Unmarshal(jsonUpdates, &opts)
+	if err != nil {
+		return gitlab.UpdateIssueOptions{}, err
+	}
+	return opts, nil
+}
+
+func QueryIssues(client *gitlab.Client, query map[string]any) error {
+	opts, err := buildListIssuesOptions(query)
+	if err != nil {
+		return err
+	}
+	options := []gitlab.RequestOptionFunc{}
+
+	for {
+		issues, resp, err := client.Issues.ListIssues(&opts, options...)
+		if err != nil {
+			return err
+		}
+
+		if len(issues) == 0 {
+			fmt.Println("No results found.")
+		}
+
+		for _, issue := range issues {
+			err := listIssueOutput(client, issue)
+			if err != nil {
+				return err
+			}
+		}
+
+		if resp.NextLink == "" {
+			break
+		}
+
+		options = []gitlab.RequestOptionFunc{
+			gitlab.WithKeysetPaginationParameters(resp.NextLink),
+		}
+
+	}
+	return nil
+}
+
+func ListProjectIssues(client *gitlab.Client, projectID string, query map[string]any) error {
+	opts, err := buildListProjectIssuesOptions(query)
+	if err != nil {
+		return err
+	}
+	options := []gitlab.RequestOptionFunc{}
+
+	for {
+		issues, resp, err := client.Issues.ListProjectIssues(projectID, &opts, options...)
+		if err != nil {
+			return err
+		}
+
+		for _, issue := range issues {
+			err := listIssueOutput(client, issue)
+			if err != nil {
+				return err
+			}
+		}
+
+		if resp.NextLink == "" {
+			break
+		}
+
+		options = []gitlab.RequestOptionFunc{
+			gitlab.WithKeysetPaginationParameters(resp.NextLink),
+		}
+
+	}
+	return nil
+}
+
+func GetIssueDetails(client *gitlab.Client, projectID string, issueIID int) error {
+	this, _, err := client.Issues.GetIssue(projectID, issueIID)
+	if err != nil {
+		return err
+	}
+	if err := listIssueOutput(client, this); err != nil {
+		return err
+	}
+
+	notes, _, err := client.Notes.ListIssueNotes(projectID, issueIID, &gitlab.ListIssueNotesOptions{})
+	if err != nil {
+		log.Fatalf("Failed to fetch issue notes: %v", err)
```

Can we just return this error and have the top-level error handling do it?